lib/ansible/modules/cloud/vmware/vmware_datastore_host.py

```diff
@@ -0,0 +1,328 @@
+    - NFS host serving nfs datastore.
+    - Required for nfs datastore type
+    - Unused for others types
+  nfs_path:
+    description:
+    - Resource path on NFS host.
+    - Required for nfs datastore type
+    - Unused for others types
+  nfs_ro:
+    description:
+    - ReadOnly or ReadWrite mount.
+    - Required for nfs datastore type
+    - Unused for others types
+    default: False
+  vmfs_device_name:
+    description:
+    - Name a the device to use as VMFS datastore
+    - Required for vmfs datastore type
+    - Unused for others types
+  vmfs_version:
+    description:
+    - VMFS version to use for datastore creation
+    - Required for vmfs datastore type
+    - Unused for others types
+  esxi_hostname:
+    description:
+    - ESXi hostname to mount the datastore.
+    required: true
+  state:
+    description:
+    - "present: Mount datastore on host if it's absent else do nothing."
+    - "absent: Umount datastore if it's present else do nothing."
+    default: present
+    choices:
+    - present
+    - absent
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Mount VMFS datastores to ESXi
+  vmware_datastore_host:
+      hostname: '{{ vcenter_hostname }}'
+      username: '{{ vcenter_user }}'
+      password: '{{ vcenter_pass }}'
+      datacenter_name: '{{ datacenter }}'
+      datastore_name: '{{ item.name }}'
+      datastore_type: '{{ item.type }}'
+      vmfs_device_name: 'naa.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+      vmfs_version: 6
+      esxi_hostname: '{{ inventory_hostname }}'
+      state: present
+  delegate_to: localhost
+
+- name: Mount NFS datastores to ESXi
+  vmware_datastore_host:
+      hostname: '{{ vcenter_hostname }}'
+      username: '{{ vcenter_user }}'
+      password: '{{ vcenter_pass }}'
+      datacenter_name: '{{ datacenter }}'
+      datastore_name: '{{ item.name }}'
+      datastore_type: '{{ item.type }}'
+      nfs_server: '{{ item.server }}'
+      nfs_path: '{{ item.path }}'
+      nfs_ro: no
+      esxi_hostname: '{{ inventory_hostname }}'
+      state: present
+  delegate_to: localhost
+  with_items:
+      - { 'name': 'NasDS_vol01', 'server': 'nas01', 'path': '/mnt/vol01', 'type': 'nfs'}
+      - { 'name': 'NasDS_vol02', 'server': 'nas01', 'path': '/mnt/vol02', 'type': 'nfs'}
+
+- name: Remove/Umount Datastores from ESXi
+  vmware_datastore_host:
+      hostname: '{{ vcenter_hostname }}'
+      username: '{{ vcenter_user }}'
+      password: '{{ vcenter_pass }}'
+      datacenter_name: '{{ datacenter }}'
+      datastore_name: NasDS_vol01
+      esxi_hostname: '{{ inventory_hostname }}'
+      state: absent
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import connect_to_api, vmware_argument_spec
+
+
+class VMwareDatastoreHost(object):
```

Comment:
Can you inherit `PyVmomi` class ? located `from ansible.module_utils.vmware import PyVmomi`